### PR TITLE
Fixed client certificate thumbprint badly reported in device twin

### LINKED ISSUE #538: Bug: Client certificate Thumbprint badly reported in Device Twin
## Expected Behavior
image

## Current Behavior
image

## Context (Environment)

**Portal version**: 1.0.7
**LoRaWAN Stack version**: 2.0

## Additional Information
image

### PR DESCRIPTION
**Pull request by audserraCGI**
## Description

What's new?

- Fix #538 
- Added Thumbprint format validation

## What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Tests
- [ ] Other

### CODE REVIEW

File: src/AzureIoTHub.Portal/Client/Validators/ConcentratorValidator.cs [hunk 1]
```diff
@@ -24,6 +24,9 @@ public ConcentratorValidator()
                 .NotEmpty().WithMessage("LoraRegion is required.")
                 .NotNull().WithMessage("LoraRegion is required.");
 
+            _ = RuleFor(x => x.ClientThumbprint)
+                .Matches("^([A-F0-9]{2}:){19}[A-F0-9]{2}$")
```
Comment (kbeaugrand) [inline src/AzureIoTHub.Portal/Client/Validators/ConcentratorValidator.cs:28]: It miss here the validation for the occurence number of tokens

File: src/AzureIoTHub.Portal/Server/Helpers/DeviceHelper.cs [hunk 2]
```diff
@@ -96,6 +96,37 @@ public static void SetDesiredProperty(Twin twin, string propertyName, object val
...
+        public static string RetrieveClientThumbprintPropertyValue(Twin twin, string propertyName)
+        {
+            ArgumentNullException.ThrowIfNull(twin, nameof(twin));
+
+            var camelCasedPropertyName = propertyName.ToCamelCase();
+
+            return twin.Properties.Desired.Contains(camelCasedPropertyName) ?
+                twin.Properties.Desired[camelCasedPropertyName][0] : null;
+        }
+
```
Comment (kbeaugrand) [inline src/AzureIoTHub.Portal/Server/Helpers/DeviceHelper.cs:129]: Why did you add this methods ? 
Why not use normal method instead ?

**Review by kbeaugrand — CHANGES_REQUESTED** [review]

File: src/AzureIoTHub.Portal/Server/Helpers/DeviceHelper.cs (hunk 2 above)
Comment (audserraCGI) [reply src/AzureIoTHub.Portal/Server/Helpers/DeviceHelper.cs:129]: I created two new methods here (get/set) because ClientThumbprint is stored as an array in the device twin, whereas other properties simply are strings.

Comment (kbeaugrand) [reply src/AzureIoTHub.Portal/Server/Helpers/DeviceHelper.cs:129]: For the name (Camel or Pascal case), we should trust in the parameter name given in the method.

For the value (string or array, or object, we may think about serializing the value in Json before setting the value to the twin instead of creating a new method ...

File: src/AzureIoTHub.Portal/Client/Validators/ConcentratorValidator.cs (hunk 1 above)
Comment (kbeaugrand) [reply src/AzureIoTHub.Portal/Client/Validators/ConcentratorValidator.cs:28]: Nevermind...

File: src/AzureIoTHub.Portal/Client/Pages/LoRaWAN/Concentrator/CreateConcentratorPage.razor [hunk 3]
```diff
@@ -92,6 +95,15 @@
     private MudForm form;
     private ConcentratorValidator concentratorValidator = new ConcentratorValidator();
 
+    public PatternMask maskThumbprint = new PatternMask("XX:XX:XX:XX:XX:XX:XX:XX:XX:XX:XX:XX:XX:XX:XX:XX:XX:XX:XX:XX")
+    {
+            MaskChars = new[] { new MaskChar('X', @"[0-9a-fA-F]") },
```
Comment (kbeaugrand) [inline src/AzureIoTHub.Portal/Client/Pages/LoRaWAN/Concentrator/CreateConcentratorPage.razor:100]:
```suggestion
            MaskChars = new[] { new MaskChar('X', @"[0-9A-F]") },
```

**Review by kbeaugrand — CHANGES_REQUESTED** [review]

**Review by kbeaugrand — CHANGES_REQUESTED** [review]

**Review by kbeaugrand — APPROVED** [review]
